### PR TITLE
otlpmetrichttp otlptracehttp: Do not parse non-protobuf reponses

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/4717

### CODE REVIEW

File: exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go [hunk 1]
```diff
@@ -262,6 +267,14 @@ func NewHTTPCollector(endpoint string, resultCh <-chan ExportResult) (*HTTPColle
 	return c, nil
 }
 
+// WithHTTPCollectorRespondingPlainText makes the HTTPCollector to return a
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go:270]:
```suggestion
// WithHTTPCollectorRespondingPlainText makes the HTTPCollector return a
```

File: exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go [hunk 2]
```diff
@@ -382,6 +395,13 @@ func (c *HTTPCollector) respond(w http.ResponseWriter, resp ExportResult) {
 		return
 	}
 
+	if c.plainTextResponse {
+		w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+		w.WriteHeader(http.StatusOK)
+		_, _ = io.WriteString(w, "OK")
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go:401]:
```suggestion
		_, _ = w.Write([]byte("OK"))
```

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go [hunk 3]
```diff
@@ -166,8 +166,7 @@ func (c *client) UploadMetrics(ctx context.Context, protoMetrics *metricpb.Resou
 			if _, err := io.Copy(&respData, resp.Body); err != nil {
 				return err
 			}
-
-			if respData.Len() != 0 {
```
Comment (MrAlias) [inline exporters/otlp/otlpmetric/otlpmetrichttp/client.go:170]: Should this not continue to also check the response length is greater than zero?

Comment (pellared) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:170]: If the server returns `Content-Type: application/x-protobuf` it to have the content.

File: exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go (hunk 2 above)
Comment (pellared) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go]: https://github.com/open-telemetry/opentelemetry-go/pull/4719/commits/af08f24e7942b7a141103846aae088821b598095

File: exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go (hunk 1 above)
Comment (pellared) [reply exporters/otlp/otlpmetric/otlpmetricgrpc/internal/otest/collector.go]: https://github.com/open-telemetry/opentelemetry-go/pull/4719/commits/af08f24e7942b7a141103846aae088821b598095

File: exporters/otlp/otlpmetric/otlpmetrichttp/client.go (hunk 3 above)
Comment (MrAlias) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:170]: I don't think that is a guarantee. An empty response if still going to be valid with those header values.

I would continue to include this guard.

Comment (pellared) [reply exporters/otlp/otlpmetric/otlpmetrichttp/client.go:170]: I added https://github.com/open-telemetry/opentelemetry-go/pull/4719/commits/46d0335ea63438d29d107d401c2a8edf773d5027

It occurs that this condition is even covered by existing test suite 🎉

**Review by dashpole — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

**Pull request merged**